src/createClient.ts

```diff
@@ -51,6 +51,15 @@ export const REPOSITORY_CACHE_TTL = 5000;
  */
 export const GET_ALL_QUERY_DELAY = 500;
 
+/**
+ * The default number of milliseconds to wait before retrying a rate-limited
+ * `fetch()` request (429 response code). The default value is only used if the
+ * response does not include a `retry-after` header.
+ *
+ * The API allows up to 200 requests per second.
+ */
+const DEFUALT_RETRY_AFTER_MS = 1000;
```

Comment:
Couldn't/doesn't the API return proper rate limit headers consistently so we can base the timeout on those?

In all fairness though there's no real standard apart from this draft: https://www.ietf.org/archive/id/draft-ietf-httpapi-ratelimit-headers-07.html, been using GitHub's on my services~

Reply:
In the planned update, the API will only return an additional `Retry-After` header. It will always return `0`, but I asked the platform team if it could return the number of milliseconds until the request can be retried successfully (as the header's name implies).

See the internal Slack discussion here: https://prismic-team.slack.com/archives/C030DCL4JNR/p1692824110830889

In the current code, that means the request will be retried immediately since it uses `0` as `setTimeout()`'s delay. `DEFAULT_RETRY_AFTER_MS` is only used if the header is missing for some reason.

If we cannot get `Retry-After` to have millisecond precision, we should at least update the value to `1` (seconds) or `1000` (milliseconds) so that the client is properly retry.

Reply:
FYI, the API now returns the number of milliseconds until the rate limit is reset via the `Retry-After` header. 🙂